File: sudoku-common/common-web/src/main/java/com/zumin/sudoku/common/web/log/Log.java
package com.zumin.sudoku.common.web.log;

import static java.lang.annotation.ElementType.METHOD;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 自定义操作日志记录注解
 */
@Target(METHOD)
@Retention(RetentionPolicy.RUNTIME)
@Documented
public @interface Log {

  /**
   * 日志标题
   */
  String value() default "";

  /**
   * 功能
   */
  BusinessType businessType() default BusinessType.INSERT;

  /**
   * 是否保存参数数据
   */
  boolean isSaveParameterData() default true;
}

File: sudoku-common/common-web/src/main/java/com/zumin/sudoku/common/web/log/Log.kt
package com.zumin.sudoku.common.web.log

/**
 * 自定义操作日志记录注解
 */
@MustBeDocumented
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class Log(
  /**
   * 日志标题
   */
  val value: String = "",
  /**
   * 功能
   */
  val businessType: BusinessType = BusinessType.INSERT,
  /**
   * 是否保存参数数据
   */
  val isSaveParameterData: Boolean = true,
)

/**
 * 业务操作类型
 */
enum class BusinessType {
  OTHER,
  INSERT,
  UPDATE,
  DELETE,
  SELECT,
  SAVE
}